Fix: Handle Missing rsun_ref in Metadata and Improve Differential Rotation Handling (#7953)
sunpy/sunpy: changelog/7953.bugfix.rst

> Fixed handling of 'rsun_ref' if it is missing in the input map in `~sunpy.physics.differential_rotation.differential_rotate`.

Reviewer: ```suggestion
Fixed a bug in :func:`~sunpy.physics.differential_rotation.differential_rotate` that assumed the input map header had ``RSUN_REF`` defined.
```

Reviewer: Also, add a newline at the end

> Fixed handling of 'rsun_ref' in `sunpy.physics.differential_rotation.py`,ensuring robust and flexible processing of solar data.

Reviewer 2: ```suggestion
Fixed handling of 'rsun_ref' if it is missing in the input map in `~sunpy.physics.differential_rotation.differential_rotate`.
```
Not sure if that link is correct.
